db/database_test.go

```diff
@@ -1631,3 +1631,228 @@ func BenchmarkPut(b *testing.B) {
+}
+
+func mockOIDCOptionsWithMultipleProvidersCBQ() *auth.OIDCOptions {
+	differentProvider := "SalesForce"
+	provider1 := mockOIDCProvider()
+	provider2 := mockOIDCProviderWithCallbackURLQuery()
+	providers := auth.OIDCProviderMap{provider1.Name: &provider1}
+	provider2.Name = "Youtube"
+	providers[provider2.Name] = &provider2
+	return &auth.OIDCOptions{DefaultProvider: &differentProvider, Providers: providers}
+}
+
+func mockOIDCOptionsWithNoIss() *auth.OIDCOptions {
+	provider := mockOIDCProviderWithNoIss()
+	providers := auth.OIDCProviderMap{provider.Name: &provider}
+	return &auth.OIDCOptions{DefaultProvider: &defaultProvider, Providers: providers}
+}
+
+func mockOIDCOptionsWithNoClientID() *auth.OIDCOptions {
+	provider := mockOIDCProviderWithNoClientID()
+	providers := auth.OIDCProviderMap{provider.Name: &provider}
+	return &auth.OIDCOptions{DefaultProvider: &defaultProvider, Providers: providers}
+}
+
+func mockOIDCOptionsWithNoValidationKey() *auth.OIDCOptions {
+	provider := mockOIDCProviderWithNoValidationKey()
+	providers := auth.OIDCProviderMap{provider.Name: &provider}
+	return &auth.OIDCOptions{DefaultProvider: &defaultProvider, Providers: providers}
+}
+
+func mockOIDCOptionsWithBadName() *auth.OIDCOptions {
+	provider := mockOIDCProvider()
+	providers := auth.OIDCProviderMap{provider.Name + "_": &provider}
+	return &auth.OIDCOptions{DefaultProvider: &defaultProvider, Providers: providers}
+}
+
+func TestNewDatabaseContextWithOIDCProviderOptionErrors(t *testing.T) {
+	tests := []struct {
+		inputOptions  *auth.OIDCOptions
+		expectedError string
+	}{
+		// Provider should be skipped if no issuer is not provided in OIDCOptions. It should throw the error
+		// "OpenID Connect defined in config, but no valid OpenID Connect providers specified". Also a warning
+		// should be logged saying "Issuer and ClientID required for OIDC Provider - skipping provider"
+		{
+			inputOptions:  mockOIDCOptionsWithNoIss(),
+			expectedError: "OpenID Connect defined in config, but no valid OpenID Connect providers specified",
+		},
+		// Provider should be skipped if no client ID is not provided in OIDCOptions. It should throw the error
+		// "OpenID Connect defined in config, but no valid OpenID Connect providers specified". Also a warning
+		//	should be logged saying "Issuer and ClientID required for OIDC Provider - skipping provider"
+		{
+			inputOptions:  mockOIDCOptionsWithNoClientID(),
+			expectedError: "OpenID Connect defined in config, but no valid OpenID Connect providers specified",
+		},
+		// If the provider name is illegal; meaning an underscore character in provider name is considered as
+		// illegal, it should throw an error stating OpenID Connect provider names cannot contain underscore.
+		{
+			inputOptions:  mockOIDCOptionsWithBadName(),
+			expectedError: "OpenID Connect provider names cannot contain underscore",
+		},
+	}
+
+	for _, tc := range tests {
```

Comment:
It would be useful to run each of these cases as a subtest, so we can easily identify which one is failing. Same in `TestNewDatabaseContextWithOIDCProviderOptions` too.

It requires adding a name to each test case.

```golang
t.Run(tc.name, func(t *testing.T) {
```

Reply:
Sure Ben, I've committed the changes in latest revision.